src/audio/module_adapter/module/volume/volume.c

```diff
@@ -1112,6 +1117,9 @@ static int volume_process(struct processing_module *mod,
 
 		/* copy and scale volume */
 		cd->scale_vol(mod, &input_buffers[0], &output_buffers[0], frames);
+		processed_bytes = frames * frame_byte;
+		audio_stream_consume(source, processed_bytes);
+		audio_stream_produce(sink, processed_bytes);
```

Comment:
Are the above consume/produce operations expensive? I think it in addition to wrap also calculates avail and free. Maybe the module adapterization could be reverted for cd->scale_vol() and us the buffer pointers from called function instead. The other components could be confused from buffer usage going back and forth.